python/pyarrow/tests/test_orc.py

```diff
@@ -171,7 +171,26 @@ def test_orcfile_empty(datadir):
     assert table.schema == expected_schema
 
 
-def test_readwrite(tmpdir):
+def test_filesystem_uri(tmpdir):
+    from pyarrow import orc
+    table = pa.table({"a": [1, 2, 3]})
+
+    directory = tmpdir / "data_dir"
+    directory.mkdir()
+    path = directory / "data.orc"
+    orc.write_table(table, str(path))
+
+    # filesystem object
+    result = orc.read_table(path, filesystem=fs.LocalFileSystem())
+    assert result.equals(table)
+
+    # filesystem URI
+    result = orc.read_table(
+        "data_dir/data.orc", filesystem=util._filesystem_uri(tmpdir))
```

Comment:
Is it also possible to do `orc.read_table(util._filesystem_uri(path))`?

Reply:
I think so. That’s a test @dongjoon-hyun added which I moved. I will add that one and see whether it still works.

Reply:
Yes it does.